edition/voxel_tool.cpp

```diff
@@ -324,6 +324,39 @@ void VoxelTool::smooth_sphere(Vector3 sphere_center, float sphere_radius, int bl
 	}
 }
 
+void VoxelTool::do_surface(Vector3 sphere_center, float sphere_radius, float strength) {
+	ZN_PROFILE_SCOPE();
+	ZN_ASSERT_RETURN(sphere_radius >= 0.01f);
+	ZN_ASSERT_RETURN(strength >= 0.0f && strength <= 1.0f);
+
+	const Box3i voxel_box = Box3i::from_min_max(
+			math::floor_to_int(sphere_center - Vector3(sphere_radius, sphere_radius, sphere_radius)),
+			math::ceil_to_int(sphere_center + Vector3(sphere_radius, sphere_radius, sphere_radius)));
+
+	// TODO Perhaps should implement `copy` and `paste` with `VoxelBufferInternal` so Godot object wrappers wouldn't be
+	// necessary
+	Ref<gd::VoxelBuffer> buffer;
+	buffer.instantiate();
+	buffer->create(voxel_box.size.x, voxel_box.size.y, voxel_box.size.z);
+
+	if (_channel == VoxelBufferInternal::CHANNEL_SDF) {
+		// Note, this only applies to SDF. It won't affect voxel texture data.
+		
+		copy(voxel_box.pos, buffer, (1 << VoxelBufferInternal::CHANNEL_SDF));
+
+		std::shared_ptr<VoxelBufferInternal> smooth_buffer = make_shared_instance<VoxelBufferInternal>();
```

Comment:
std::shared_ptr is not necessary, VoxelBufferInternal can be created on the stack